### PR TITLE
Refactor API to look like native bevy API

### LINKED ISSUE #3: Refactor API to look like native bevy API

### PR DESCRIPTION
**Pull request by bytemunch**
Attempts to fix #3 

Makes the cosmic-text creation process look like:

```rs
// spawn a new CosmicEditBundle
commands.spawn(CosmicEditUiBundle {
    style: Style {
        // Size and position of text box
        width: Val::Px(300.),
        height: Val::Px(50.),
        left: Val::Px(100.),
        top: Val::Px(100.),
        ..default()
    },
    ..default()
});
```

Not made a PR in forever lol, lmk if there's anything else I should include.

### CODE REVIEW

**Comment by Dimchikkk** [comment]
Hi @bytemunch, thanks for the PR! The code changes look good to me, I want to integrate changes to `velo` project and test the solution before merging, let you know once the changes are tested.

File: src/lib.rs [hunk 1]
```diff
@@ -7,63 +9,351 @@ use bevy::{
...
+pub struct CosmicAttrs(pub AttrsOwned);
+
+impl Default for CosmicAttrs {
+    fn default() -> Self {
+        CosmicAttrs(AttrsOwned::new(Attrs::new()))
+    }
+}
+
+#[derive(Component, Default)]
+pub struct CosmicBackground(Option<Handle<Image>>);
```
Comment (Dimchikkk) [inline src/lib.rs]:
```suggestion
pub struct CosmicBackground(pub Option<Handle<Image>>);
```

**Comment by Dimchikkk** [comment]
@bytemunch I haven't tested on `velo` (have to find time to move `velo` to new API) but it works nicely on examples, I am eager to merge it. API definitely improved, thanks!

**Comment by bytemunch** [comment]
You're very welcome, good pickup on that background image :)

**Review by Dimchikkk — APPROVED** [review]

**Pull request merged**